src/reset-password/ResetPasswordPage.jsx

```diff
@@ -0,0 +1,143 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { Button, Input, ValidationFormGroup } from '@edx/paragon';
+import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
+
+import messages from './messages';
+import { resetPassword, validateToken } from './data/actions';
+import { resetPasswordResultSelector } from './data/selectors';
+import InvalidTokenMessage from './InvalidToken';
+import ResetSuccessMessage from './ResetSuccess';
+import Spinner from './Spinner';
+
+
+const ResetPasswordPage = (props) => {
+  const { intl } = props;
+
+  const [newPasswordInput, setNewPasswordValue] = useState('');
+  const [confirmPasswordInput, setConfirmPasswordValue] = useState('');
+  const [passwordValid, setPasswordValidValue] = useState(true);
+  const [passwordMatch, setPasswordMatchValue] = useState(true);
+
+  const handleNewPasswordChange = (e) => {
+    const newPassword = e.target.value;
+    setNewPasswordValue(newPassword);
+    const isValid = newPassword.length >= 8 && newPassword.match(/\d+/g);
+    setPasswordValidValue(isValid !== false);
+  };
+  const handleConfirmPasswordChange = (e) => {
+    const confirmPassword = e.target.value;
+    setConfirmPasswordValue(confirmPassword);
+    setPasswordMatchValue(confirmPassword === newPasswordInput);
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    if (newPasswordInput === '') {
+      setPasswordValidValue(false);
```

Comment:
You also need to return after this, otherwise it will call the action because the `passwordValid` state will stay `true` until the the react reloads the component. Or change it to if/else.